### PR TITLE
173 / Update createNew and startExisting workout pages

### PR DESCRIPTION
**Pull request by L-Hen**
Replaced nav bar with newest version. Implemented selection to createNew exercises. Implemented functional search bar.
Cleaned up index pages.

# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [x] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [ ] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?
Implemented functionality to "start new exercise" page.

# Tests

How was this tested?

- [ ] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [ ] NO documentation needed

Link to external documentation:

# [Optional] Are there any post-deployment tasks we need to perform?
Still need functionality
Filter by equipment (Button)
Use Diagram (Button)
Send array of exercises selected to following page.
Start Freeform Workout (Button)

### CODE REVIEW

**Comment by thienguen** [comment]
reviewing now

File: apps/expo/src/app/workout/createNew/index.tsx [hunk 1]
```diff
@@ -3,11 +3,12 @@ import {StyleSheet, View, Text, TouchableOpacity, TextInput, ScrollView } from '
 import { router } from 'expo-router'
 import { SafeAreaView } from 'react-native-safe-area-context'
 import ExerciseList from '~/components/workout/exerciseList'
-
+import NavBar from '~/components/ui/nav-bar/NavBar'
+import Nav from '~/app/nav'
```
Comment (thienguen) [inline apps/expo/src/app/workout/createNew/index.tsx:7]: remove unused import

Comment (thienguen) [reply apps/expo/src/app/workout/createNew/index.tsx:7]: also, run prettier format on this,

File: apps/expo/src/app/workout/createNew/index.tsx [hunk 2]
```diff
@@ -17,42 +18,15 @@ export default function createNew() {
...
-          <Text className="rounded-[5px] text-[20px] text-[#FFFFFF] text-center">
-            Use Diagram
-          </Text>
-        </TouchableOpacity>
-      </View>
-      <View style={{height: 1, backgroundColor: '#CACACA', marginHorizontal: 13, opacity: 0.3,}}/>
-      <ScrollView className="mb-[25px] py-[5px]">
-        <ExerciseList />
-      </ScrollView>
+
```
Comment (thienguen) [inline apps/expo/src/app/workout/createNew/index.tsx:26]: image

the filter by equipment and the use diagram need to be synced.

Also we run into errors of

image

I believe Elliot pr fix this. or I could be mistaken, regardless, its an error

File: apps/expo/src/app/workout/startExisting/index.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import {StyleSheet, View, Text, TouchableOpacity, TextInput, ScrollView } from '
 import { router } from 'expo-router'
```
Comment (thienguen) [inline apps/expo/src/app/workout/startExisting/index.tsx:3]: remove unsused import, format prettier x2

File: apps/expo/src/app/workout/startExisting/index.tsx [hunk 4]
```diff
@@ -18,18 +18,11 @@ export default function createNew() {
```
Comment (thienguen) [inline apps/expo/src/app/workout/startExisting/index.tsx:18]: there is a lot of questionable code in this file. :( 
we also run into the same problem of 

```
 ERROR  Warning: Each child in a list should have a unique "key" prop.
 ```

Comment (thienguen) [reply apps/expo/src/app/workout/startExisting/index.tsx:18]: image

why there is 2 workout list,  

```
<View style={{height: 1, backgroundColor: '#CACACA', marginHorizontal: 13, opacity: 0.3, top: 10, }}/>
```

that view doesn't do anything.

File: apps/expo/src/components/workout/exerciseList.tsx [hunk 5]
```diff
@@ -4,6 +4,8 @@ import { Text, View } from 'react-native'
 import { ScrollView } from 'react-native-gesture-handler'
```
Comment (thienguen) [inline apps/expo/src/components/workout/exerciseList.tsx:5]: prettier x3

File: apps/expo/src/app/workout/createNew/index.tsx (hunk 2 above)
Comment (thienguen) [reply apps/expo/src/app/workout/createNew/index.tsx:26]:
```
        <TouchableOpacity className="px-4 py-[6px] rounded-[5px] fixed border-[#CACACA] bg-[#424242]" >
            <Text className="rounded-[5px] text-[20px] text-[#FFFFFF] text-center">
              Start Freeform Workout
            </Text>
        </TouchableOpacity>
```

I can't see where this button is at all 

image

**Review by thienguen — CHANGES_REQUESTED** [review]
yea, naw

**Comment by AaronFmyHub** [comment]
I got a warning say add unique "Key" Prop, but a warning is a warning

File: apps/expo/src/app/workout/createNew/index.tsx [hunk 6]
```diff
@@ -17,42 +18,15 @@ export default function createNew() {
...
-            Next
-          </Text>
-        </TouchableOpacity>
-      </View>
-      <View className="mx-3 border-[1px] border-[#CACACA] bg-[#272727] rounded-[5px]">
-        <TextInput
-          className="text-[12px] text-[#CACACA] px-4 py-[6px] mx-1
-          placeholder:text-[#CACACA] placeholder:text-[20px] placeholder:italic" 
-          placeholder= "Search workout by name..."/>
+        <NavBar
```
Comment (AaronFmyHub) [inline apps/expo/src/app/workout/createNew/index.tsx:21]: Have less room at top
image

**Comment by thienguen** [comment]
@mva919 Please suggest your changes to the layout of this page

File: apps/expo/src/app/workout/createNew/index.tsx (hunk 1 above)
Comment (L-Hen) [reply apps/expo/src/app/workout/createNew/index.tsx:7]: done

**Comment by L-Hen** [comment]
Addressed comments, ready for re-review

**Comment by thienguen** [comment]
> Addressed comments, ready for re-review

re-review by tonight

**Comment by thienguen** [comment]
image

**Comment by thienguen** [comment]
the above was fine, 

but could have some padding so it doesn't stick to the rear too much like what you did to the other page

image

good work implementing the taylor navbar

**Review by thienguen — APPROVED** [review]

**Pull request merged**